dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/AbstractAnalyticsService.java

```diff
@@ -102,7 +104,21 @@ protected Grid getGrid( EventQueryParams params )
 
         for ( QueryItem item : params.getItems() )
         {
-            grid.addHeader( new GridHeader( item.getItem().getUid(), item.getItem().getDisplayProperty( params.getDisplayProperty() ), item.getValueType(), item.getTypeAsString(), false, true, item.getOptionSet(), item.getLegendSet() ) );
+            if ( item.getValueType() == ValueType.ORGANISATION_UNIT
+                && params.getCoordinateField().equals( item.getItem().getUid() ) )
+            {   // Special case: if the request contains an item of Org Unit value type and the item uid
+                // is linked to coordinates (coordinateField), then create an Header of ValueType
+                // COORDINATE and type "Point"
+                grid.addHeader( new GridHeader( item.getItem().getUid(),
+                    item.getItem().getDisplayProperty( params.getDisplayProperty() ), COORDINATE,
+                    Point.class.getName(), false, true, item.getOptionSet(), item.getLegendSet() ) );
```

Comment:
This feels a bit "iffy". Could we instead change it so that the QueryItem contains the correct value type? We do have `ValueType.COORDINATE`. If the Query item contains this value type in the case of org unit, we can remove this if statement.

Reply:
Do we only support Point in analytics? Org Units can be any type of Geometry, right? While Coordinates is always a Point?

Reply:
I have some back and forth with Bjorn regarding this: Org Units can be Point, but also Polygons etc. So, there were two possibilities: only return data for OrgUnit which use Point Geometry or use the "Centroid" function in Postgres, which returns the center of a geometry. That's what we decided to use: we always return a type "Point" and when we query the db we use the `ST_Centroid`, which will always return a Point, no matter what geomtery the OU has.

Reply:
@larshelge I agree 100% but it's a bit tricky to do it differently: I'll explain. If I do for example something like that when we map the http request to the actual params:
```
final QueryItem queryItem = getQueryItem( dim, pr, request.getOutputType() );
                    
if ( queryItem.getValueType() == ValueType.ORGANISATION_UNIT
    && request.getCoordinateField().equals( queryItem.getItem().getUid() ) )
{
    queryItem.setValueType( COORDINATE );
}
```
If I do something like this, then I "loose" the knowlege that the attribute was actually ORG UNIT. When I get to the point where I need to build the SQL query, I need to know the actual column "_geom", but since the query item is now a coordinate type I will not be able to do that. I hope it makes sense. Pretty sure there is some other way, just can't see it now. Lets have a quick chat when you have a second?